### ISSUE TITLE
some questions

### ISSUE THREAD

**Issue by taki0112** [HsinYingLee/DRIT#10]
1. why do not use mean in [kl_loss](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L298)
why sum ? 
in [other kl_loss](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L387).. you use mean ! 

2. why 2 discriminator(disA, disA2) [link](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L45)
what do you think just update it once with batch size 1?

3. why did you calculate latent regression loss using ***only mu***?
[link](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L183)
[link2](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L362)
According to the paper, should we calculate the random_normal distribution using `mu and stddev` and calculate the loss?

4. According to the paper, KL_loss seems to fit only for the attribute, why does the code calculate the KL_loss for the content? [link](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L306)

**Comment by taki0112** [comment]
[link](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L183)
Why did you calculate latent regression loss using ***only mu***?
According to the paper, should we calculate the random_normal distribution using `mu and stddev` and calculate the loss?

**Comment by taki0112** [comment]
https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L306

**Comment by hytseng0509** [comment]
1. For [kl_loss](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L298), please refer to the official Pytorch [example](https://github.com/pytorch/examples/blob/6fd43cd04f4677be005167157c848f0fefe131af/vae/main.py#L84) and Appendix B from VAE [paper](https://arxiv.org/pdf/1312.6114.pdf).
2. For [other kl_loss](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L387), it is just an L2 regularization loss, not a KL divergence loss. We'll change the confusing variable name.
3. From the observation stated in Section 4 of the BicycleGAN [paper](https://arxiv.org/pdf/1711.11586.pdf), using separate discriminators yields better visual results.
4. For latent regression, since the goal is to recover back to a single randomly sampled point in attribute latent space, we constrain the predicted `mu` with an L1 loss.
5. Sorry for the confusing loss name. It is just an L2 regularization loss, not a KL divergence loss. We will change the loss name.

**Comment by taki0112** [comment]
OK. 
Here is my [tensorflow implementation](https://github.com/taki0112/DRIT-Tensorflow)

Let me know if there's an error in my code.
Thank you for sharing your good research.
